[GOV] proposed code of conduct update - mature sktime structure and decision making on resources (#3095)
sktime/sktime: docs/source/get_involved/code_of_conduct.rst

is expected to show respect and courtesy to other community members at
all times.

The sktime Community Council, and all project members,

> are dedicated to a **harassment-free experience for

Reviewer: this is good

Author: well, thanks!

1 Introduction
--------------

sktime is a community-oriented and -led project.

We value the involvement of everyone in the community, and strive to be an open
community. We commit to provide fair and revarding opportunities for everyone without
restriction, to participate, learn, and become leaders of the community. We expect from
members of our community to create such opportunities for others as have been created

> for them.

Author: interesting question. I would say: it should be an expectation, according to engagement and "power". I.e., a new contributor or junior career person is not expected to since it is not in their power to. A very senior academic or industry data scientist who wants to have "power" in `sktime` (e.g., on the CC, or core developer) is expected to mentor, donate, or create opportunities otherwise, in transparent and non-exclusionary manner.

> for them.

Reviewer 2: I tend to agree with what Lovkush is suggesting here though.  I agree it should be a community standard/supported and value in the community, but it doesn't quite feel like a code of conduct thing? Or something that could generate a Code of Conduct violation.  I think doing so leads to a situation that can be hard to equitably enforce without structured oversight/could run the risk of undermining other more "traditional" sorts of complaints people might raise as CoC violations?

Perhaps there is another way to communicate this though? ie spelled out more clearly in terms of what you sign up for by becoming a core dev or CC member?

> for them.

Author: > ie spelled out more clearly in terms of what you sign up for by becoming a core dev or CC member?

hm, yes, I think that might be an option

> I agree it should be a community standard/supported and value in the community, but it doesn't quite feel like a code of conduct thing? 

Hm, I *do* think it is a code of conduct thing. The situation I have in mind is the "negative" of it, where someone hijacks community assets for their personal network, actively prevents opportunities out of turf mentality, keeps certain people "out" due to base motivation (e.g., chauvinism).

Systemic versions of racism and sexism nowadays don't take the form of communication (which the old CoC focuses on mostly), but denial of resource, denial of opportunities, shrouded in plausible deniability.

The systemic versions appear when we talk about substantial resource allocation decisions, in the order that motivates this CoC change.

Example of racism, "old CoC type": open name calling in a social media thread; talking over people in a meeting. That is very communication focused.

Example of racism, "new CoC type": advertising a position in a way that people outside socially connected circles, with certain nationalities (outside affluent countries) or outside non-diverse "ingroup networks" will have a substantially harder time to find the opportunity, apply for it, or succeed, assuming equal level of competence.

Plausible deniability: "oh, admin says we have to do X" or "I did advertise"

Silent exclusionary measures: tailoring the job description in a very specific way, advertising only in low visibility outlets, deferring exclusion to the set-up of the position (e.g., causing problematic visa requirements), restricting timelines to make applicaion outside direct network difficult.

None of this is caught well by the "old style CoC", because all communication was extremely polite.

> for them.

Author: Important: the motivation of systemic exclusion is typically hard to spot and near impossible to prove.

*but*, the exclusionary measures are easy to spot and can be put in a code of conduct as forbidden. Hence the discussion of paper trail, transparency, etc.

> for them.

Author: In fact, I think code of conduct may be the only tool we have to regulate intentional or accidental of systemic exclusion as discussed above.

> for them.

Reviewer 2: Basically I still think if it is going to be something that you want to be enforcable then it needs to be a bit more specific (and people need to opt in) otherwise I still think it should be something that is under community culture/guidelines and/or core dev agreements.  I very much agree that "lack of opportunities" and other related issues you bring up here are very important and certainly can impact someone's experience with `sktime` and software development more broadly, however I don't necessarily think that is the point here.  To me the question is - what is the best way to encourage this within the community? 

I don't believe that putting it in the Code of Conduct and making it a potentially enforceable offence is the way to go.  Here are my reasons why:
- As a mostly volunteer organization people will inevitably be involved in sktime to various degrees, how do you quantify how much each individual should be giving to the community/creating opportunities?  Ultimately unless you are specific there will be conflicts, because a persons assessment of their own skill, commitment amount, and seniority will not perfectly line up with anyone else's. On top of that it can be tricky to compare or measure how much someone is contributing to the community/creating opportunities...
- It stands apart from the other expectations in the Code of Conduct because rather than asking people *not* to do something it is asking people *to* do something.  Which again I think is hard to expect/enforce in a volunteer community. Maybe it is something that you can ask of core devs or people on CC/CoC, but I think it should be framed as an opt-in responsibility rather than a general Code of Conduct thing. 
- Lastly, because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly.  I do think this runs the risk of alienating certain individuals if they are not on board with this change, and I think it runs the risk of making the CoC as a whole be seen as a document that can be used for political issues, especially if it boils down to situations like you allude to above of plausible deniability vs CoC....  I am not saying that is the intent, I just can imagine it being seen that way, and I think it could undermine the underlying message/goal.

However, if you were able to make some of these expectations more concrete then I could be in favor.  For example:
 - if you are creating an opportunity that makes use of `sktime` resources then that opportunity should be shared and advertised with all of the sktime community. 
 - if you are open to mentoring individuals within the `sktime` community then you should open that opportunity up for anyone who wants to engage/apply!  
 - etc etc

> for them.

Author: > It stands apart from the other expectations in the Code of Conduct because rather than asking people not to do something it is asking people to do something

Hm, is that really so? The CoC has both positives and negatives already. Entire section "expected behaviour" - "be respectful", "use inclusive language", etc, these are positive asks "to do sth" in my reading?

> because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly.

Of course - every expectation needs to be as specific as possible, or can be used as "rubber conditions" arbitrarily. The legal principle here is [legal certainty](https://en.wikipedia.org/wiki/Legal_certainty).

We need something unambiguous, I think the negative examples in section 2.2 are pretty good in being concrete.

Question: what is so different between

"written or verbal comments which have the effect of excluding people on the basis of protected characteristics"

and 

"decisions on resource allocation or operations which have the effect of excluding people on the basis of protected characteristics"

(note: in the current condition focused on communication, intention is also not a condition for it to apply!)

> Lastly, because of the above vague nature of expectations/results/whether or not people have opted in I do worry that it could be used (or be seen to be used) unfairly. I do think this runs the risk of alienating certain individuals if they are not on board with this change

I think there are two issues in this sentence:

* worry that the rules will be used unfairly
* worry whether everyone is onboard with the rules

These are two things, we should separate them!

"used unfairly": fairness must be built in the design, I agree. We can borrow principles from legal theory. Number one is legal certainty (see above), conditions must not be vague. Number two is also important imo - separation of powers. The CoC committee that judicates should be separate from the CC, and, optimally, unconnected to any network.
This will not prevent political use (nor does it in "real life"), but the political influence in judication is minimized.

"everyone is onboard": there is a chicken/egg problem here. The document is the very definition of what the community considers as "good" or "fair". Formally, the CC can decide this, not the entire community needs to be on board.

This might sound harsh, but let me explain by example my stance on this.
In a hypothetical situation, there surely exist, somewhere, people who like making racist or sexist jokes a lot and consider that kind of environment as positive ("boys club atmosphere"). I have been told one doesn't need to look to far in tech...

Personally, I would not shead a single tear, a person like that can leave. My vision of `sktime` is one of a cohesive, but professional, respectful, and diverse community.
Same thing for people who hire only friends and disregard merit or fair opportunities. Or people who can't be convinced to commit to even the most elementary anti-corruption measures and transparency requirements. In a sense that's just the systemic version of the "boys club".
One can't help that standing up against that will inevitably be framed as "political" or "toxic" or similar. It's also sad times we live in that an anti-corrupt stance is not broad consensus, but considered "political".

> for them.

Reviewer 3: > and transparency requirements

Several months ago, I put suggestion forward that all core devs ought to reveal any affiliations and how their efforts are paid for (e.g. voluntary, part of phd, part of job, etc.).  This is to help community know if there are any major outside influences and to prevent any 'takeovers'.  I think only Franz and I liked the suggestion (despite the impetus for this being that both Franz and I (as well as other contributors) were at Shell, so I thought it was useful for community to be aware of this).

I dont think anybody explained why they disliked it. But I never pushed for an explanation either.

Sad indeed that striving for best practice is 'political'! But that is the way of the world.

> for them.

Reviewer 2: I do agree that whatever you and the CC committee decide can/should be incorporated into this document. :)

In truth I kind of thought/hoped this was more a discussion about the *best* way to encourage people to make more opportunities within `sktime` and make the community more open/diverse!  Not sure why it felt like we kept having a discussion about whether or not "boys club's" or sexist jokes should be condoned...

But at the end of the day I really think we are on the same page with goals and if leadership think this is the best way to go then I am all for it! :)  In truth this is probably a small point and the discussion over it has probably grown out of proportion. 

Interesting that the disclosure of ties didn't go over well?  Maybe it could be something worth bringing up again in the future.

> for them.

Author: numfocus guidance is to (a) disclose affiliation for CC members, and (b) contractual stipulation that no two members can be from the same group/entity.

We could make it a requirement for CC members, but I suppose that's a separate discussion.

I have added now a section on best practice and transparency.

We are committed to scientific quality and strive to be an open forum on matters
of technical content and open community governance. All community members are expected
to adhere to the principle of free speech in academia and science, in order to create

> the best possible science and the best possible open community.

Reviewer 3: more a comment than any suggestion, but I think there is a fundamental tension between 'best possible science' and 'open community'.

- in my mind, 'best possible science' implies high standards which cannot reasonably be expected from the majority of contributors (or the majority of professional scientists in fact. top quality science is fundamentally hard!)
- there are personality traits (e.g. high anxiety or low self-esteem) which can make it difficult and very uncomfortable for people to receive critical feedback on a contribution - even if the criticism is constructive and would help them improve. This means there is trade-off between how high quality you want that individual contribution to be, versus, how welcome to you want that person to feel.

I do not know if there a solution to this. My own preference is to lean more towards being welcoming at the cost of 'best possible science'

Author: I think it's one of the trade-offs. We should be welcoming to early career contributors, and don't disadvantage them at all if something is not perfect. Conversely, we can reasonably expect of early career contributors to take on mentoring opportunities offered to them, and invest time in learning best practice, before wanting to radically change `sktime` core designs.
(e.g., everyone can be expected to read up on linting or how to use conda. We're happy to give pointers repeatedly and explain at lengths in the communtiy session, but we are not forced to merge PR that don't get over the code quality boundaries)

Reviewer 3: Happy with explanation. Maybe worth rewording this sentence to make clear that we do not expect high absolute skills levels.

Author: yes, reworded it now that it's to the best of our abilities:

> to the best of their abilities, contribute to jointly creating
the best possible science and the best possible open community.

We are committed to creating a friendly and respectful place for learning,
teaching and contributing. All participants in our in-person events and online
communications are expected to show respect and courtesy to others at all times.

To make clear what is expected, everyone participating in activities
associated with the sktime project is required to conform to this Code
of Conduct. This Code of Conduct applies to all spaces managed by the
sktime project including, but not limited to, in-person focus groups and
workshops, and communications online via GitHub.

The communicty council of sktime (see current composition on "roles")
is responsible for defining the Code of Conduct,
and can be contacted under info@sktime.org.

> The Code of Conduct Committee of sktie is responsible for enforcing the Code of Conduct.

Reviewer 3: ```suggestion
The Code of Conduct Committee of sktime is responsible for enforcing the Code of Conduct.
```

Reports can be made via the sktime issue tracker, and will be kept confidential.

2 Code of Conduct
-----------------

The sktime team are dedicated to providing a welcoming and supportive
environment for all people, regardless of background or identity. As
such, we do not tolerate:
* behaviour that is disrespectful to our
  community members or that excludes, intimidates, or causes discomfort to others.
* discrimination or harassment based on protected characteristics, including
  restriction of fair opportunities based on protected characteristics
* silencing of reasonable opinions and stifling of free speech on topics within
  the project scope (science and open community operations)

In the rare cases where our ideals are in conflict, we strive to protect,
in the context of power dynamics, the less powerful party or parties.

> Protected characteristics as referred to above include, but are not limited to:

Reviewer 3: adding quotes
```suggestion
'Protected characteristics' as referred to above include, but are not limited to:
```

to adhere to the principle of free speech in academia and science, and, to the best
of their abilities, contribute to jointly creating
the best possible science and the best possible open community.

We are committed to creating a friendly and respectful place for learning,
teaching and contributing. All participants in our in-person events and online
communications are expected to show respect and courtesy to others at all times.

To make clear what is expected, everyone participating in activities
associated with the sktime project is required to conform to this Code
of Conduct. This Code of Conduct applies to all spaces managed by the
sktime project including, but not limited to, in-person focus groups and
workshops, and communications online via GitHub.

The Code of Conduct Committee of sktime is responsible for defining and
enforcing the Code of Conduct, according to Section 4 below.

Incidents can be reported to the Code of Conduct Committee as outlined in Section 3, and
are handled in accordance with Sections 3 and 4.

The community council of sktime (see current composition on "roles")
is responsible for ensuring that resolutions of the Code of Conduct committee

> are adhered to, and can be contacted under info@sktime.org.

Reviewer: what does this mean, its a bit cryptic. Maybe just reformulate?

Author: "responsible" as per RACI, technical term.

Do you have a suggestion how to better formulate this?

enforcing the Code of Conduct, according to Section 4 below.

Incidents can be reported to the Code of Conduct Committee as outlined in Section 3, and
are handled in accordance with Sections 3 and 4.

The community council of sktime (see current composition on "roles")
is responsible for ensuring that resolutions of the Code of Conduct committee
are adhered to, and can be contacted under info@sktime.org.

Reports may be reviewed by other members of the core development team,

> unless there is a conflict of interest, and will be kept confidential.

Reviewer: I would not remove this bcz its less democratic then, so can you kindly put that in again?

Author: I removed it because there was no process, e.g., who decides who may view what.

I agree with your comment - removing the ability for core devs to view the paper trail makes the process less transparent and the CoC commitee less accountable.

So, I will add it back but define the process.

Author: added this:

```
4.6 Audit of Investigation Process
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

At any point in time, core developers may request the number of ongoing and concluded
investigations over the last year before the date of the query.
Any group of three core developers may also request access to all investigation reports
in a time period, or reports to cases with specified properties.
If there is no conflict of interest (as determined by the Code of Conduct Committee),
access will be shared to anonymized reports for a period of 2 weeks, and the reports
must not be copied or otherwise proliferated.
```

Author: the clause

```
Reports of code of conduct committee case proceedings will be kept confidential.
```

is now in section 4.3 (where it belongs imo)

* discrimination or harassment based on protected characteristics, including
  restriction of fair opportunities based on protected characteristics
* silencing of reasonable opinions and stifling of free speech on topics within

>   the project scope (science and open community operations)

Reviewer: I dont get what this part could be concretely, so I would either make it more clear or remove it. Like this it can be anything? What is "silencing", just expressing another opinion might already then be "silencing" another opinion?

Author: > What is "silencing", just expressing another opinion might already then be "silencing" another opinion?

No, that is not the case as per intention.

I can see how the formulation is vague, i.e., what is "silencing", what is "reasonable". I removed that.

-  **Be respectful of different viewpoints and experiences**. Do not
   engage in homophobic, racist, transphobic, ageist, ableist, sexist,
   or otherwise exclusionary behaviour.

>    In disagreements, assume best intentions.

Reviewer: what is this concretely? Sorry but it sounds a bit like random "bla bla"? :)

Author: Same as "be respectful of different viewpoints". This is a guiding principle, examples are spelled out in appendix A and further below.

-  nonconsensual or unwelcome physical contact
-  sustained disruption of talks, events or communications
-  insults or put downs
-  sexist, racist, homophobic, transphobic, ableist, or exclusionary jokes
-  excessive swearing
-  incitement to violence, suicide, or self-harm
-  continuing to initiate interaction (including photography or
   recording) with someone after being asked to stop
-  publication of private communication without consent
-  decision making on resource allocation which has the effect of excluding people
   on the basis of protected characteristics, or other forms of systemic discrimination

> -  decision making on resource allocation that confers advantages to personal networks,

Reviewer 2: On the topic of resource allocation I was wondering - does `sktime` have any sort of formal budget?  I was unable to find such a document, but wonder if it could be a good thing to put into place going forward! :) Reasons being:
- It can be helpful in securing future funding if you show evidence of where money has been spent in the past/how.
- Can help the community achieve certain goals if where the money will be spent is planned/discussed. 
- Is often helpful in reducing conflict in advance if such a document is in place already!

Could be something the community could get together and discuss once or twice a year?  Could also be an opportunity for brainstorming and/or grant writing sessions? 

Just chiming in on this as I know this year `sktime` paid for the dev-days and helped the GSoC interns come! (Which I am of course biased in favor of having been a good idea!) and I think having some sort of paper trail that tracks these sorts of activities can be helpful.  Would other people find this a useful exercise going forward?  (Or does it perhaps already exist and I have missed it?)

Reviewer 3: I am generaly in favour of more open-ness and having some kind of record/paper trail.

Regarding budget/ This page has some info on funding: https://www.sktime.org/en/stable/about/funding.html.  i didn't know this, but just saw there is a link to where you can donate to sktime: https://opencollective.com/sktime.  and then if you explore, you see how the opencollective funds are spent: https://opencollective.com/sktime#category-BUDGET

Author: > does sktime have any sort of formal budget?

yes, and no, multiple things:

On status:
* `sktime` is not a formal legal entity, so all budgets are currently "on behalf of the project", but the formal status is usually at different places, and accountability is with different individuals affiliated with `sktime`
* we are currently affiliated with NumFOCUS and exploring to move to sponsored status. That would give `sktime` a status as a formal legal entity (part of umbrella NumFOCUS), and NumFOCUS would become the budget administering entity
* the Alan Turing Institute is not related to `sktime` status, that we are here has more historical reasons

On de-facto budgets:
* `sktime` has an "open collective" account where we currently administer donations and reimbursements https://opencollective.com/sktime, this is fully open.
* `sktime` has recently won an UKRI grant of approx 500.000 GBP volume: https://gow.epsrc.ukri.org/NGBOViewGrant.aspx?GrantRef=EP/W030756/1. This is administered by individuals at UEA, one of our partner universities. Said individuals are `sktime` affiliated, but formally have full control and do not need to (and choose not to) provide transparent views of what the money is being spent on. `sktime` itself is not a formal entity, and not an independent research organisation, so cannot apply for grants. I think this mismatch of accountability/authority is a fatal construction flaw in the UKRI grant call that was meant to support open source communities. Similar issues exist with funds sourced from US and continental European government research funding agancies.
* there are further accounts that individuals administer to the benefit, or on behalf of `sktime`. For instance, I have a small pot of emergency funds reserved for `sktime` at UCL, some funds that originate from my consultancy work 5 years ago and which I can use at my discretion to the benefit of `sktime` (or any other academic activities).
* as above, the Alan Turing Institute does not offer any de-facto assistance to `sktime`. Despite us being here (and figuring in public facing presentation on the great work the Turing is doing), we are not receiving any benefits from the Turing. We are also not involved or benefitting from the substantial funding pots allocated to "open research community building", see here: https://github.com/alan-turing-institute/open-community-building. Personally, I find that strange, but formally the Turing do not *need* to support `sktime` in any way, and they choose not to.

Author: so, from the perspective of code of conduct, we need to set out expectations for a complex system of individual stakeholders, `sktime` affiliated.
Some might get awarded benefits and resources due to achievements of the `sktime` commnunity, and, once awarded, typically have almost absolute authority on decision making on said resources. Some others might choose to donate to `sktime` or award benefits to the community, but said benefits may benefit some parts of the community and not others.

-  nonconsensual or unwelcome physical contact
-  sustained disruption of talks, events or communications
-  insults or put downs

> -  sexist, racist, homophobic, transphobic, ableist, or exclusionary jokes

Reviewer: I think that part was already mentioned above under "protected characteristics"? but I dont mind to have it here again

Author: this is unchanged and has been in the CoC since the start. It only appears as a change because I removed a newline.

-  excessive swearing
-  incitement to violence, suicide, or self-harm
-  continuing to initiate interaction (including photography or
   recording) with someone after being asked to stop
-  publication of private communication without consent
-  decision making on resource allocation which has the effect of excluding people
   on the basis of protected characteristics, or other forms of systemic discrimination
-  decision making on resource allocation that confers advantages to personal networks,
   and excludes others, e.g., preventing advertising of opportunities, not applying best practice in recruitment
-  reducing transparency of operations of sktime, e.g., by obfuscation of the paper trail
-  invoking the code of conduct dishonestly, or as a form of bullying,
   especially from a position of power

> Some more principled guidance is given in the appendix "Guidance in applying the CoC".

Reviewer: can you kindly remove that part, this is in my opinion a very controversial section. What is the definition of "resource"? In my opinion we have above clarified already enough the the community  values and what kind of behavior we appreciate and do not. Why would that even require to be repeated here for "resource allocation"? Its cleat that those values are applied at any point in time and to any decision making process.

Author: > can you kindly remove that part, this is in my opinion a very controversial section.

This is the key part of the PR!

> Its cleat that those values are applied at any point in time and to any decision making process.

Am I correctly summarizing your point, that you think this is redundant since it follows from the community principles?

If you think it is redundant, I think it makes sense to list these: the section collects examples of unacceptable behaviour for the benefit of clarity.

The original purpose of this section is to have common misbehaviours called out as examples, as to preempt discussion on how the community principles are to be applied.

Or, do you think that the new examples are acceptable behaviour? Would appreciate some clarity here.

3.1 Contact points
~~~~~~~~~~~~~~~~~~

Please contact the current chair of the Code of Conduct Committee directly by email, see 4.1.

3.2 Alternate contact points
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In case of conflict of interest of sktime CoC Committee members, you can report

> through the `NumFOCUS CoC process <https://numfocus.org/code-of-conduct>`__,

Reviewer: is that actually approved by NumFOCUS? I think the did not even answer our last mails?

Author: in the last meeting we had with them at the CC, they said that we should not default to them, but they would act as a "last resort" if we give them jurisdiction.

My memory is a bit hazy (this PR is from July) and can't find the minutes at the moment, do you remember the date or a keyword for searching?

3.1 Contact points
~~~~~~~~~~~~~~~~~~

> Please report CoC incidents via the [reporting tool in the issue tracker](https://github.com/alan-turing-institute/sktime/issues/new/choose).

Reviewer 2: I was curious about how the reporting tool worked, so opened the link and the page doesn't seem to exist?  If we are still in the process of building it - I had some suggestions/ideas as to what form this could take:
 - what if instead of just making a CoC incident issue tracker we use an (optionally anonymous) feedback form, which contains within it the functionality for anyone to report a CoC incident (which would of course then be flagged for more stringent triage)?  Emphasis on it being a *feedback form* over a CoC incident reporting tool as something I have seen work well in communities I have been a part of is the form can also serve to collate issues/ideas/suggestions from the community that can be addressed at the next community meeting.  It also makes it open to all sorts of feedback (ie positive feedback or new ideas) instead of just CoC complaints!  Of course there can be drawbacks to anonymity, but in my experience I have found that it sometimes lowers the bar to entry and can encourage more feedback from the community.
 - We could implement such a form via Google Forms - which could then be shared with all core contributors/a outside third party for review. 
 - If concerned about misuse we could remove the option for anonymity for a formal CoC complaint, but this could potentially keep the door open for people to provide feedback on sktime community and experience without needing to escalate to a CoC complaint? 
 
 If interested I can provide a mock-up of what a form could look like on Google Forms?

Author: >I was curious about how the reporting tool worked, so opened the link and the page doesn't seem to exist?

It should exist! Maybe this got destroyed somehow when we migrated documentation and no one checked 😢 
I think as a take-away, we should add, on the responsibility list of the CoC committee to perform regular checks of the reporting workflows and ensure they are kept up to date. Perhaps asking for a demonstration of a clickthrough every 6 months, to the CC.

> what if instead of just making a CoC incident issue tracker we use an (optionally anonymous) feedback form, which contains within it the functionality for anyone to report a CoC incident (which would of course then be flagged for more stringent triage)? Emphasis on it being a feedback form

Excellent idea! I am in favour

> If concerned about misuse

I think, in the balance of all things, misuse is less likely than genuine reports that want to remain anonymous. Even if more likely, I personally would weigh the trade-off strongly in favour of the latter.

> If interested I can provide a mock-up of what a form could look like on Google Forms?

Useful, in faour! What do you think, @aiwalter, @Lovkush-A ?

Where did the old reporting form end up being? @mloning, do you still know where it was/should be?

Author: @martinagvilas, do you know where the CoC reporting form is? See above, it seems the reporting tool link is now broken.

Reviewer 2: [Here is an example](https://docs.google.com/forms/d/e/1FAIpQLSdaIpWJ2bAnkBBuYwCrAyzixHJ_n23IxkcdB6zJvFvW2_u_Ug/viewform?usp=sf_link) of what the form could look like - very short/simple.
  I went ahead and made both @fkiraly and @Lovkush-A editors, but happy to share that with anyone else who is interested as well.  I was thinking we could add the form to the end of all official correspondences (ie to the signature of the emails from the sktime toolbox?) and could also setup a slack bot that invites people to share feedback once a month? (or maybe on a similar frequency as the core dev meetings?).  In my experience such forms don't necessarily get used too heavily, but its nice to have them around! :)

Reviewer 3: I like the short concise nature of form. Main comment is to make clear that there are two possible places the feedback can be used:

- For CoC issues, then only seen by CoC group.
- For everything else, discussed by core devs.

Author: generally, every form needs to say the purpose if is used for and the entity/recipient that will have a view of contents

Reviewer 2: In that case I think what makes the most sense is:
- the results of this form are only visible to the CoC committee.  
- they are responsible for sharing all responses that don't indicate that they are CoC violations with the rest of the core dev team during meetings? 

I will add those changes in addition to a more specific explanation of who will read the responses/in what context to the above form shortly! :)

Author: yes, that makes sense. From a technical standpoint, I suppose we then need a gmail email that is shared by CoC committee? Or did you imagine this differently? 

Not sure about gmail workings, but there needs to be a "main" gmail acct for all google forms, right? Or wrong?

Reviewer 2: Yea!  I went ahead and make you the owner Franz (since I am not on the CoC committee don't really think it makes sense for me to have access to this form once it goes live!).  

From a technical perspective I agree the best way to do it could be to have a shared mailing list (gmail based) that is for the CoC committee.  I think this would make it easier to add/remove people as needed.  and I imagine there would be a way to set up regular digests/updates to people who are subscribed when the form is updated?

Author: opened an issue here to set this up "properly".
https://github.com/sktime/sktime/issues/3664

Apologies for leaving this for a while...
I started my new job on Aug 3 and through that my capacity (and attention) got much reduced.

> Please report CoC incidents via the [reporting tool in the issue tracker](https://github.com/alan-turing-institute/sktime/issues/new/choose).

Reviewer: ```suggestion
Please report CoC incidents to info@sktime.org.
```

Author: I would actually introduce more "separation of powers", one flaw (in my opinion) of the current construction is the overlap between CC and CoC committee, due to potential conflicts of interest at leadership level.

In an ideal world, I think they should be staffed by distinct individuals, and have no overlap whatsoever.

grateful for `this guidance from Ada
Initiative <http://geekfeminism.wikia.com/wiki/Conference_anti-harassment/Responding_to_reports>`__.

We highly appreciate the work that all of the communities linked above
have put into creating code of conduct documents and processes.

This Code of Conduct is licensed under a `Creative Commons Attribution
4.0 International <https://creativecommons.org/licenses/by/4.0/>`__ (CC
BY 4.0 CA) license which means you are free to share and adapt the work
so long as attribution is maintained to substantial sources of contribution:

* The `Django project <https://www.djangoproject.com/>`__
* The `Contributor Covenant <http://contributor-covenant.org/>`__,
* The `Citizen Lab <https://citizenlab.ca/>`__
* The `Alan Turing Institute Data Study Groups <https://www.turing.ac.uk/collaborate-turing/data-study-groups>`__ organising team
* `The Turing Way <https://github.com/alan-turing-institute/the-turing-way>`__ project
* `The Carpentries <https://carpentries.org/>`__
* The ``sktime`` project
* other sources and influences mentioned above

> Appendix A: Guidance in applying the CoC

Reviewer: I have reviewed until here, will continue another time. Its quite a lot text below which seems to contain quite some controversial sections, so factoring this out might be the best. Especially there is no definition of "resource" and as I wrote already above, why do we need to even have a section on this?

Author: Thanks for reading!

I would appreciate if you spell out explicitly whether you yourself approve or disapprove.

E.g., which sentences do *you* (Martin) want to have struck, changed, etc.

Author: I will add a definition of "resource" in appendix A

Examples for in-scope:

* an academic allocating grant monies from a grant, or a data science consultant providing consultancy services, with a primary ``sktime`` branding
* hiring decisions for roles that are advertised with a primary ``sktime`` branding
* spending decisions from ``sktime`` community administered accounts
* decisions arising from academic collaboration or commercial requests directed to the ``sktime`` community,
  e.g., via official email, or via digital ``sktime`` discussion tools (slack, GitHub discussion etc)

Best practice
^^^^^^^^^^^^^

To ensure decision making in line with the code of conduct, decision making must be:

* by the ``sktime`` community, through ``sktime`` community decision making mechanisms.
  E.g., following decision making outlined in the ``sktime`` governance document.
* in communication transparent to the ``sktime`` community, following communication requirements outlined in the ``sktime`` governance document.
* pursuant of and in line with the charitable mission of ``sktime`` and NumFOCUS.
  For-profit activities must feed back the majority of benefits obtained through it
  to the ``sktime`` community and its charitable entities.

> 

Reviewer 4: Cannot wrap my head around the "majority of benefits" regulation, how should that be measured? Could ultimately be very intimidating if people fear having to transfer majority of earnings.

"feed back a significant amount of benefits proportionate to the involvement of sktime in the activity" or something like that would be less intimidating

Author: I removed the clause since it seems to cause controversy.
(removed on the open PR https://github.com/sktime/sktime/pull/3674)

As discussed on Friday, it is indeed unclear how to measure "majority of benefits".

Instead, this should be indirectly ensured by making this subject to sktime decision making mechanisms and transparency.
